geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSet.java

```diff
@@ -98,6 +102,156 @@ public static boolean del(Region<ByteArrayWrapper, RedisSet> region,
     }
   }
 
+  public static int scard(Region<ByteArrayWrapper, RedisSet> region, ByteArrayWrapper key) {
+    RedisSet redisSet = region.get(key);
+    if (redisSet != null) {
+      return redisSet.size();
+    } else {
+      return -1;
+    }
+  }
+
+  public static boolean sismember(Region<ByteArrayWrapper, RedisSet> region,
+      ByteArrayWrapper key, ByteArrayWrapper member) {
+    RedisSet redisSet = region.get(key);
+    if (redisSet != null) {
+      return redisSet.contains(member);
+    } else {
+      return false;
+    }
+  }
+
+  public static Collection<ByteArrayWrapper> srandmember(Region<ByteArrayWrapper, RedisSet> region,
+      ByteArrayWrapper key, int count) {
+    RedisSet redisSet = region.get(key);
+    if (redisSet != null) {
+      return redisSet.srandmember(count);
+    } else {
+      return null;
+    }
+  }
+
+  public static Collection<ByteArrayWrapper> spop(Region<ByteArrayWrapper, RedisSet> region,
+      ByteArrayWrapper key, int popCount) {
+    RedisSet redisSet = region.get(key);
+    if (redisSet != null) {
+      return redisSet.doSpop(region, key, popCount);
+    } else {
+      return null;
```

Comment:
does the calling code differentiate between null and an empty collection? If not, can we just return an empty collection and get rid of the upstream null checks.

Reply:
done